executor/wasm-host/src/host.rs

```diff
@@ -1485,6 +1486,158 @@ pub fn casper_env_block_time<S: GlobalStateReader, E: Executor>(
     Ok(block_time.value())
 }
 
+#[derive(Clone, Copy)]
+pub struct EnvInfo {
```

Comment:
`#[repr(C)]` if we will transmute it back and forth? (this probably wouldn't do any difference to layout as it's made of PODs, but still)

Also, this could be safely put in wasm-common crate so we can also access this from SDK

Do you you need to pass entity_addr_ptr? Balance check can be done with casper_env_balance anyway. I'm worried about ux - getting env data (block time,transferred value etc - values not specific to an entity, and also mixing data specific for an entity)